src/sections/Question/index.tsx

```diff
@@ -1,7 +1,196 @@
+    name: '',
+    email: '',
+    message: '',
+  });
+  const [nameStatus, setNameStatus] =
+    useState<InputLayoutProps['status']>(undefined);
+  const [emailStatus, setEmailStatus] =
+    useState<InputLayoutProps['status']>(undefined);
+  const [messageStatus, setMessageStatus] =
+    useState<InputLayoutProps['status']>(undefined);
+
+  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { name, email, message } = inputs;
+    if (validateForm()) {
+      contactMessage(
+        name.trim(),
+        email.trim(),
+        message.trim(),
+        (err: AxiosError, message: AxiosResponse) => {
+          if (err) {
+            // Error
+            setMessageStatus({
+              state: 'error',
+              text: 'An error occurred - please try again later',
+            });
+            setIsButtonDisabled(false);
+          } else {
+            // Success
+            setNameStatus({
+              state: 'success',
+            });
+            setEmailStatus({
+              state: 'success',
+            });
+            setMessageStatus({
+              state: 'success',
+              text: 'Your message has been sent!',
+            });
+            setIsButtonDisabled(true);
+          }
+        }
+      );
+    } else {
+      setIsButtonDisabled(true);
+    }
+  };
+
+  const validateForm = (updateStatus: boolean = true) => {
+    const { name, email, message } = inputs;
+    return (
+      validateName(name, updateStatus) &&
+      validateEmail(email, updateStatus) &&
+      validateMessage(message, updateStatus)
+    );
+  };
+
+  const validateName = (name: string, updateStatus: boolean = true) => {
+    if (name.trim().length == 0) {
+      if (updateStatus) {
+        setNameStatus({
+          state: 'error',
+          text: 'Please enter your name',
+        });
+      }
+      return false;
+    }
+    setNameStatus(undefined);
+    return true;
+  };
+
+  const validateEmail = (email: string, updateStatus: boolean = true) => {
+    if (!validator.isEmail(email)) {
+      if (updateStatus) {
+        setEmailStatus({
+          state: 'error',
+          text: 'Please enter a valid email',
+        });
+      }
+      return false;
+    }
+    setEmailStatus(undefined);
+    return true;
+  };
+
+  const validateMessage = (message: string, updateStatus: boolean = true) => {
+    if (message.trim().length == 0) {
+      if (updateStatus) {
+        setMessageStatus({
+          state: 'error',
+          text: 'Please enter your message',
+        });
+      }
+      return false;
+    }
+    setMessageStatus(undefined);
+    return true;
+  };
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const id = e.target.id;
+    const value = e.target.value;
+    if (id == 'name') {
+      validateName(value);
+    } else if (id == 'email') {
+      validateEmail(value);
+    } else if (id == 'message') {
+      validateMessage(value);
+    }
```

Comment:
Put this in a useEffect

Reply:
I'm confused, what would be in the dependencies array?

Reply:
whoops i wasnt very specific (sowie). Move the validation stuff to a useEffect and the deps would be all the states related to the inputs

Reply:
The states related to the inputs call the useEffect on first render, which triggers the validation stuff and displays errors (since inputs are all initially empty). What should I do in that case?

Reply:
we can check for dirty, but its not worth the effort... then I guess you're right and we would just stick with this implementation since it works